### PR TITLE
Ashank Kumar

### PR DESCRIPTION
**Pull request by ak3962**
Ashank Kumar, programs vowels, sum, and unique. And the assignment 4 questions.

### CODE REVIEW

File: sum.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* Ashank Kumar, user inputs an array, proram outputs cumulative sum of the elements in the array so far */
+#include <stdio.h>
+
+int main()
+{
+	int x;
+	int total = 0;
+
+	printf("Enter the length of the string:");
```
Comment (oldclesleycode) [inline sum.c:9]: Perfect job! The only thing is I think you meant to write "length of the array" but that's not a big deal at all

File: unique.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+				goto end_nested;
+			}
+
+		}
+	}
+
+	end_nested:
+	if(valid)
+		printf("The string is unique\n");
+	if(string[y] == string[z])
```
Comment (oldclesleycode) [inline unique.c:39]: Nicely done, but make sure to test for edge cases! Empty string is a unique string.

File: vowels.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
+/*Ashank Kumar, takes the vowels out of a string and prints the new string*/
+#include <stdio.h>
+#include <string.h>
+
+int main()
```
Comment (oldclesleycode) [inline vowels.c:5]: Your code prints extra characters after every output. It works, but has these extra characters you didn't clean up.

File: Assignment4.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Ashank Kumar
+
+1. strcat(string1, string2). This line of code adds two two strings together, the second string to the end of the first string..
```
Comment (oldclesleycode) [inline Assignment4.txt:3]: Right!

File: Assignment4.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+2. The size of the array is determined once data has been inputted
```
Comment (oldclesleycode) [inline Assignment4.txt:5]: right, so what would happen if it isn't?

File: Assignment4.txt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+3. int array[8][8] = {}; This line of code creates an 2d array with 8 rows, with each row having 8 columns.
```
Comment (oldclesleycode) [inline Assignment4.txt:7]: Right!

File: Assignment4.txt [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+4a. strcmp() compares two strings ASCII values. If the first string is larger a positive number that is the difference between the strings ASCII values will be returned. The same will happen if the second string is larger, but the number will be negative. If they are equal a 0 will be returned.
```
Comment (oldclesleycode) [inline Assignment4.txt:9]: Perfect! 💯

File: Assignment4.txt [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+4b. fgets() reads strings from the keyboard by using the name of a variable, its size, and the command stdin
```
Comment (oldclesleycode) [inline Assignment4.txt:11]: Right!

File: Assignment4.txt [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+4c. strcar concatenates or adds on string to the end of another.
```
Comment (oldclesleycode) [inline Assignment4.txt:13]: Right!

File: Assignment4.txt [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+4d. Returns the length of a string excluding the \0
```
Comment (oldclesleycode) [inline Assignment4.txt:15]: Yup!

File: vowels.c [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+			continue;
+		}
+		else
+		{
+			change[counter] = string[x];
+			counter++;
+		}
+			
+	}
+
```
Comment (oldclesleycode) [inline vowels.c:59]: So the random symbols printed out because of the following: you're just missing one line that would make your line perfect! Remember that a string has to have a null terminator at the end! because you forgot to make the last element one, it printed out those junk symbols! 

So add `change[counter] = '\0';` and your problem goes away!
